### PR TITLE
Add ceilometer-polling service as required for Liberty installation

### PR DESCRIPTION
**Pull request by mmnelemane**
Services agent-central, agent-ipmi have been deprecated upstream from Kilo release onwards. These are now replaced by ceilometer-polling. The current ceilometer barclamp does not address this deprecation. Neither ceilometer-agent-central nor ceilometer-polling services run on the control node failing some ceilometer functionalities (like metering for control node services). This change updates the barclamp to install and start ceilometer-polling service instead of the deprecated services.

### CODE REVIEW

**Comment by tboerger** [comment]
It's missing a migration for the role rename.

**Comment by mmnelemane** [comment]
tboerger: Thanks. I will add the migrations.

File: crowbar_framework/app/models/ceilometer_service.rb [hunk 1]
```diff
@@ -182,9 +182,9 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     # the VIP of the cluster to be setup
     allocate_virtual_ips_for_any_cluster_in_networks(server_elements, vip_networks)
 
-    central_elements, central_nodes, central_ha_enabled = role_expand_elements(role, "ceilometer-cagent")
+    polling_elements, polling_nodes, polling_ha_enabled = role_expand_elements(role, "ceilometer-polling")
```
Comment (mmnelemane) [inline crowbar_framework/app/models/ceilometer_service.rb]: Requires openstack-ceilometer-polling binary to be installed. This is provided in SR: https://build.opensuse.org/request/show/345055 and https://build.opensuse.org/request/show/345059

File: chef/cookbooks/ceilometer/attributes/default.rb [hunk 2]
```diff
@@ -40,9 +40,9 @@
 default[:ceilometer][:api][:service_name] = api_service_name
 default[:ceilometer][:collector][:service_name] = collector_service_name
 default[:ceilometer][:agent_notification][:service_name] = agent_notification_service_name
-default[:ceilometer][:central][:service_name] = central_service_name
-default["ceilometer"]["alarm_evaluator"]["service_name"] = alarm_evaluator_service_name
-default["ceilometer"]["alarm_notifier"]["service_name"] = alarm_notifier_service_name
+default[:ceilometer][:polling][:service_name] = polling_service_name
+default[:ceilometer][:alarm_evaluator][:service_name] = alarm_evaluator_service_name
```
Comment (dirkmueller) [inline chef/cookbooks/ceilometer/attributes/default.rb]: unrelated change? if you do that, please also change the callers

File: chef/cookbooks/ceilometer/attributes/default.rb [hunk 3]
```diff
@@ -16,11 +16,11 @@
...
 default[:ceilometer][:group]="ceilometer"
 
-central_service_name = "ceilometer-agent-central"
+polling_service_name = "ceilometer-polling"
 if %w(suse).include?(node[:platform_family])
-  central_service_name = "openstack-ceilometer-agent-central"
+  polling_service_name = "openstack-ceilometer-polling"
 elsif %w(rhel).include?(node[:platform_family])
-  central_service_name = "openstack-ceilometer-central"
+  polling_service_name = "openstack-ceilometer-polling"
```
Comment (vuntz) [inline chef/cookbooks/ceilometer/attributes/default.rb]: You can change this to:
```ruby
if %w(rhel suse).include?(node[:platform_family])
  polling_service_name = "openstack-ceilometer-polling"
end
```

File: chef/cookbooks/ceilometer/recipes/polling.rb [hunk 4]
```diff
@@ -13,21 +13,21 @@
...
 #
 
-package "ceilometer-agent-central" do
+package "ceilometer-polling" do
   if %w(suse).include?(node[:platform_family])
-    package_name "openstack-ceilometer-agent-central"
+    package_name "openstack-ceilometer-polling"
   elsif %w(rhel).include?(node[:platform_family])
-    package_name "openstack-ceilometer-central"
+    package_name "openstack-ceilometer-polling"
```
Comment (vuntz) [inline chef/cookbooks/ceilometer/recipes/polling.rb]: Same here.

**Comment by vuntz** [comment]
Minor nitpicks, but looks mostly good!

**Comment by mmnelemane** [comment]
@vuntz: Thanks. Just updated the conditions.

**Comment by vuntz** [comment]
+1

**Comment by dirkmueller** [comment]
-1, you need to bump the schema-revision in the json file to 23

**Comment by mmnelemane** [comment]
@dirkmueller: Thanks. Done!!!

**Comment by dirkmueller** [comment]
+1
